vm/src/obj/objsequence.rs

```diff
@@ -12,6 +12,270 @@ use crate::function::OptionalArg;
 use crate::pyobject::{BorrowValue, PyObject, PyObjectRef, PyResult, TryFromObject, TypeProtocol};
 use crate::vm::VirtualMachine;
 
+pub trait PySliceableSequenceMut: PySliceableSequence {
```

Comment:
should this trait require PySliceableSequence? If `fn as_slice(&self) -> PySliceableSequence` is possible, we can skip `impl PySliceableSequence for Vec` which may causes unexpected cost